lib/membrane_http_adaptive_stream/manifest.ex

```diff
@@ -112,19 +87,21 @@ defmodule Membrane.HTTPAdaptiveStream.Manifest do
     )
   end
 
-  @spec finish(t, Track.id_t()) :: t
+  @spec finish(t, Track.id_t()) :: {Track.Changeset.t(), t}
   def finish(%__MODULE__{} = manifest, track_id) do
-    update_in(manifest, [:tracks, track_id], &Track.finish/1)
+    get_and_update_in(manifest, [:tracks, track_id], &Track.finish/1)
   end
 
   @doc """
-  Restores all the stale segments in all tracks.
-
-  All the tracks must be configured to be persisted beforehand, otherwise this function will raise
+  Restores all the stale segments in all tracks that have option persisted? set to true.
   """
   @spec from_beginning(t()) :: t
   def from_beginning(%__MODULE__{} = manifest) do
-    tracks = Bunch.Map.map_values(manifest.tracks, &Track.from_beginning/1)
+    tracks =
+      Bunch.Map.map_values(manifest.tracks, fn track ->
+        if Track.is_persisted?(track), do: Track.from_beginning(track), else: track
+      end)
```

Comment:
Maybe we should only return persisted tracks?

Reply:
I don't think so. In most cases, after the call, we want to serialize the manifest (for example on termination or `end_of_stream`) and it would be hard to do with half of the tracks.  What do you think?

If we want to know which ones are persisted we can call `Track.is_persisted?`

Reply:
> it would be hard to do with half of the tracks

Why so? Actually, what would be the point of keeping tracks that weren't persisted?